### ISSUE TITLE
Facebook login shows error for an existing user

### ISSUE THREAD

**Issue by ananyo2012** [NIT-dgp/cat-forum#46]
Currently the facebook login shows an error for an existing user. If you have signed up using the login form and then log in with facebook it shows an error due to duplicate record. This happens because the facebook login tries to create a new record for the user with the same email id.

**Comment by divyanshumehta** [comment]
I would like to work on this issue

**Comment by ananyo2012** [comment]
Ok sure goodluck with it. Let me give you some context. To start with facebook login is implemented using the `omniauth-facebook` gem. The code for the omniauth_callbacks_controller that governs the login is [here](https://github.com/NIT-dgp/cat-forum/blob/github_master/app/controllers/users/omniauth_callbacks_controller.rb). There is a method named `find_for_facebook_oauth` that finds or creates the user instance that s used to login. Here is the code for the [method](https://github.com/NIT-dgp/cat-forum/blob/github_master/app/models/user.rb#L34-L44). Refer to [this](http://stackoverflow.com/a/22126562/4798734) stackoverflow answer for solving this issue. You should also see the [socail-login-in-rails](https://github.com/mohitjain/social-login-in-rails) given at the end of that answer and the relevant files there for better understanding.

**Comment by divyanshumehta** [comment]
So first i would have to sign up with facebook developers and get an `app-id` and `app-secret`and there on proceed. Isn't it.

**Comment by ananyo2012** [comment]
Yes go to the fb developers site and create an app and get the app-id and
app-secret from there. You need to put them in secrets.yml
ᐧ
